### PR TITLE
Don't drain the body for non-streamed entities in EmberClient

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

**Review by armanbilge — APPROVED** [review]
Thanks, good catch :)

**Comment by valencik** [comment]
Just to double check my understanding, this is a small performance optimization not a correctness / bug fix, right?

In the case of empty or strict entities we don't need to worry about draining all the bytes from the underlying stream because they are already all in memory, we don't have the risk of leaving some bytes from the socket unread?

**Comment by armanbilge** [comment]
> In the case of empty or strict entities we don't need to worry about draining all the bytes from the underlying stream because they are already all in memory

Exactly.

https://github.com/http4s/http4s/blob/54bec9943212a51970973ff145106bfcf7881a1e/core/shared/src/main/scala/org/http4s/Entity.scala#L60-L61

File: ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClient.scala [hunk 1]
```diff
@@ -44,7 +44,12 @@ final class EmberClient[F[_]] private[client] (
   ): F[Throwable] = F match {
     case concurrentF: Concurrent[F @unchecked] =>
       implicit val C = concurrentF
-      resp.body.compile.drain.as(UnexpectedStatus(resp.status, req.method, req.uri))
+      resp.entity match {
+        case Entity.Empty | Entity.Strict(_) =>
+          F.pure(UnexpectedStatus(resp.status, req.method, req.uri))
+        case Entity.Default(body, _) =>
+          body.compile.drain.as(UnexpectedStatus(resp.status, req.method, req.uri))
```
Comment (valencik) [inline ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClient.scala:51]: The readability here will be super nice when/if https://github.com/http4s/http4s/pull/6305 merges :sunglasses:

**Review by valencik — APPROVED** [review]

**Pull request merged**